Kin-Rewards-Engine/Reputation-Model-Proposal.md

```diff
@@ -0,0 +1,44 @@
+
+1. The distribution model is both, known to all and deterministic. Any reward engine one might think of is vulnerable to attacks and abuse. Keeping the model semi-transparent will make it much more prone to these attacks. If secrets aren't desired due to fear of centralization, we would suggest adding randomness to the model. Specifically, in the model proposed, a simple bribery attack is possible - (relating to the example in the RFC file) `Daniel` could pay `Kaitlyn` a small bribe for her to transact with him, leading to him sweeping the majority of the reward that day. In addition, even if `Kaitlyn` is an innocent user, her power is overwhelming - her vote is practically the only one that matters.
+
+2. Not a one-coin-one-vote mechanism. As shown in the example some coins may not count towards the reward. This might seem ok, but it allows another abuse - it allows coins to count more than once. If, for example, `Emma` opens a new Kin wallet - `Emma2` - with 0 Kin and after doing all her transactions from the original `Emma` wallet she transfers her remaining 500 Kin to the new `Emma2` wallet and then transacts again with the ecosystem, her coins will count twice, distorting the original intent of the reward engine.
+
+
+## Proposal
+
+We believe a reputation system run by Kik is not only possible, but inevitable, at least so long as the system is in bootstrap mode and a more stable PoS mechanism is found. The reputation model is made possible due to one key observation, namely, that most users interacting with the Kin ecosystem will do it from the Kik messenger app (at least in the early days). Assuming there will be an option to get a Kin wallet within the app (or to incorporate an existing one with it), a link can be made between devices and Kin addresses. This mapping will allow a reputation model based on a one-device-one-vote basis. Here is a possible start of such a reputation system.
+
+It will apply to both users and developers (i.e., similarly to Uber's drivers' and passengers' reputation), which:
+
+1. Will be weighted during users' voting and calculation of developers' rewards.
+2. Will be used to incentivise behaviors we would like to promote, while punishing bad acting (reduce Sybil attacks, promote participation and spending, etc.).
+3. Will be individually visible to users (without the actual derivation, probably) in order to achieve the above.
+4. Everyone would be able to verify that the rewards are being distributed correctly and according to the pre-defined rules. We will periodically publish and explain what behavior increases/decreases the reputation, in order to incentivise good behavior, but won't necessarily share how is it calculated exactly.
+
+### Guidelines
+
+1. Developers will be identified entities. Their reputation will be calculated according to the quality of their KYC, the detail of their app plan, the marketing that they do, etc.
```

Comment:
Pointing out we need to put emphasis on making registration + KYC process seamless and quick, in order to not hinder economy development. Developers might be put off if they need to spend too much time and other resources on bureaucracy instead of on product.

Reply:
That's right.